### PR TITLE
Update to new interceptable location APIs

### PR DESCRIPTION
**Pull request by captainsafia**
Addresses #55152

### CODE REVIEW

**Comment by captainsafia** [comment]
@dotnet/source-build-internal Any guidance on the source build issues here?

**Comment by MichaelSimons** [comment]
I believe these two new prebuilts should be added to the SourceBuildPrebuiltBaseline as allowed prebuilts.  These won't be prebuilts in the product source-build because the live Rolsyn version will get picked up because the dependency is declared in the Version.Details.xml file.

File: src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs [hunk 1]
```diff
@@ -98,10 +102,13 @@ public Endpoint(IInvocationOperation operation, WellKnownTypes wellKnownTypes, S
...
 
     public (string File, int LineNumber, int CharacterNumber) Location { get; }
+#pragma warning disable RSEXPERIMENTAL002 // Experimental interceptable location API
+    public InterceptableLocation InterceptableLocation { get; }
+#pragma warning restore RSEXPERIMENTAL002
     public IInvocationOperation Operation { get; }
 
     public override bool Equals(object o) =>
-        o is Endpoint other && Location == other.Location && SignatureEquals(this, other);
+        o is Endpoint other && InterceptableLocation == other.InterceptableLocation && SignatureEquals(this, other);
```
Comment (RikkiGibson) [inline src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs:111]: I don't think we implemented `operator ==` on these. That might be a bug on our side. I'll have to follow up.

**Review by RikkiGibson — APPROVED** [review]

File: src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs (hunk 1 above)
Comment (captainsafia) [reply src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs:111]: An explicit `==` implementation would be nice. I assume this does implement equality checks correctly though given that it does the correct things in our cache invalidation tests for incremental generators.

**Review by MichaelSimons — APPROVED** [review]
Source-Build change LGTM.

**Comment by captainsafia** [comment]
@dotnet/aspnet-build Can I get help merging this? The macOS build is successful on AzDo but there appears to be an issue with the GH status check.

**Pull request merged**